config/courses.json

```
@@ -3,62 +3,71 @@
     "url": "https://github.com/LibreLingo/LibreLingo-ES-from-EN/archive/refs/heads/main.zip",
     "sdir": "LibreLingo-ES-from-EN-main",
     "tdir": "spanish-from-english",
+    "deploy": true,
```

oh and a last question, what's the difference between `"deploy"` and `"enable"`?

---

also, my understanding is that there's currently no difference between a course being deployed vs installed. It's just about the context, if someone is setting up their dev environment, then it's install, but during the automatic deployment process it's deploy.

Other than the naming though, it makes a lot of sense. But maybe then we can also consider the naming of the script itself:

/scripts/installExternalCourses.py  -> /scripts/setupCoursesForDeployment.py

or maybe, support 2 different use cases? Development vs deployment.

```bash
/scripts/installExternalCourses.py production  # Installs all courses that are meant to be deployed in production
/scripts/installExternalCourses.py development  # Installs all courses that are meant to be installed in a local environment
```

what do you think? I'm not sure if it makes sense. But in that case I perhaps imagine the config file having this syntax:

```suggestion
  "enabled": {
    "development": true,
    "production": true
  }
```

---

btw I remember from our video call that you mentioned that this process is missing from the documentation: https://librelingo.app/docs/

that happened because I forgot to update the documentation when this install logic was introduced. Perhaps this Pull Request is the ideal time to put this feature in the documentation.

You can change this file in order to document your script: https://github.com/LibreLingo/LibreLingo/blob/main/docs/index.md

---

dev: generate for the dev-tool (we might occasionally need to disable something even there)
deploy: put the json files on the production server
enable: show on the front-page (this is not used yet)

---

so I suggest something like this:

```json
  "enabled": {
    "development": true,
    "production": true
  }
```

just for clarity. it's better if it's self-explanatory, rather than having to document it and make people read the documentation. (Not that it cannot be mentioned in docs anyway)

---

btw I think that the same config file should be used for implementing the landing page and in general course selectors. so basically we will need to have some config also to configure whether a course should be shown in course selectors (the rest will be only shown in the dev tools)

---

That's what the "enable" was intended for.

---

perhaps `"public"` would be a better name for that? :thinking: